src/CodeStyle/Core/CodeFixes/AbstractFormattingAnalyzer.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    /// <summary>
+    /// This analyzer implements <see cref="DiagnosticAnalyzer"/>, but wraps the true formatting analyzer
+    /// implementations. Since the compiler doesn't provide the use of workspace-layer APIs for analyzers, we need to
+    /// provide our own copy of the workspace assemblies and load them using reflection if they are not already loaded
+    /// in the current process.
+    /// </summary>
+    internal abstract class AbstractFormattingAnalyzer
+        : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        internal const string FormattingDiagnosticId = "IDE0054";
```

Comment:
>FormattingDiagnosticId [](start = 30, length = 22)

All other IDEXXXX diagnostic IDs are defined in [http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/Analyzers/IDEDiagnosticIds.cs,5](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/Analyzers/IDEDiagnosticIds.cs,5), and this approach is very likely going to cause a duplicate defined in there. Can we either add a comment there about this IDE being used, or better define the ID in that file and add it to this project as a linked file?